File: src/programmingwithdocuments/workingwithimages/compressimages/java/CompressImages.java
/* 
 * Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Words. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
package programmingwithdocuments.workingwithimages.compressimages.java;

import com.aspose.words.*;

import java.io.File;
import java.net.URI;
import java.text.MessageFormat;


public class CompressImages
{
    public static void main(String[] args) throws Exception
    {
            // The path to the documents directory.
        String dataDir = "src/programmingwithdocuments/workingwithimages/compressimages/data/";
        String srcFileName = dataDir + "Test.docx";

        System.out.println(MessageFormat.format("Loading {0}. Size {1}.", srcFileName, getFileSize(srcFileName)));
        Document doc = new Document(srcFileName);

        // 220ppi Print - said to be excellent on most printers and screens.
        // 150ppi Screen - said to be good for web pages and projectors.
        // 96ppi Email - said to be good for minimal document size and sharing.
        final int desiredPpi = 150;

        // In Java this seems to be a good compression / quality setting.
        final int jpegQuality = 90;

        // Resample images to desired ppi and save.
        int count = Resampler.resample(doc, desiredPpi, jpegQuality);

        System.out.println(MessageFormat.format("Resampled {0} images.", count));

        if (count != 1)
            System.out.println("We expected to have only 1 image resampled in this test document!");

        String dstFileName = srcFileName + ".Resampled Out.docx";
        doc.save(dstFileName);
        System.out.println(MessageFormat.format("Saving {0}. Size {1}.", dstFileName, getFileSize(dstFileName)));

        // Verify that the first image was compressed by checking the new Ppi.
        doc = new Document(dstFileName);
        DrawingML shape = (DrawingML)doc.getChild(NodeType.DRAWING_ML, 0, true);
        double imagePpi = shape.getImageData().getImageSize().getWidthPixels() / ConvertUtil.pointToInch(shape.getSize().getX());

        assert (imagePpi < 150) : "Image was not resampled successfully.";
    }

    public static int getFileSize(String fileName) throws Exception
    {
        File file = new File(fileName);
        return (int)file.length();
    }
}

File: Plugins/Aspose_Words_Java_for_PHP/src/programmingwithdocuments/workingwithimages/compressimages/php/CompressImages.php
<?php
/*
 * Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Words. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
class CompressImages {
    public static function main(){
        // The path to the documents directory.
        $dataDir = "/usr/local/apache-tomcat-8.0.22/webapps/JavaBridge/Aspose_Words_Java_For_PHP/src/programmingwithdocuments/workingwithimages/compressimages/data/";
        $srcFileName = $dataDir . "Test.docx";
        $messageFormat = new Java("java.text.MessageFormat");
        $file_size = CompressImages::getFileSize($srcFileName);
        echo java_values($messageFormat->format("Loading {0}. Size {1}.", $srcFileName, $file_size));
        die("I M HERE");
        $doc = new Java("com.aspose.words.Document",$srcFileName);
        // 220ppi Print - said to be excellent on most printers and screens.
        // 150ppi Screen - said to be good for web pages and projectors.
        // 96ppi Email - said to be good for minimal document size and sharing.
        $desiredPpi = 150;
        // In Java this seems to be a good compression / quality setting.
        $jpegQuality = 90;
        // Resample images to desired ppi and save.
        $resampler = new Java("com.aspose.words.Resampler");
        $count = $resampler->resample($doc, $desiredPpi, $jpegQuality);
        echo $MessageFormat->format("Resampled {0} images.", $count);
        if ($count != 1)
            echo "<br> We expected to have only 1 image resampled in this test document!";
        $dstFileName = $srcFileName . "Resampled Out.docx";
        $doc->save($dstFileName);
        echo $messageFormat->format("Saving {0}. Size {1}.", $dstFileName, CompressImages::getFileSize($dstFileName));
        // Verify that the first image was compressed by checking the new Ppi.
        $doc = new Java("com.aspose.words.Document",$dstFileName);
        $nodeType = new Java("com.aspose.words.NodeType");
        $shape = $doc->getChild($nodeType->DRAWING_ML, 0, true);
        $convertUtil = new Java("com.aspose.words.ConvertUtil");
        $imagePpi = $shape->getImageData()->getImageSize()->getWidthPixels() / $convertUtil->pointToInch($shape->getSize()->getX());
        if($imagePpi < 150) {
            echo "Image was not resampled successfully.";
        }
    }
    public static function getFileSize($fileName) {
        $file = new Java("java.io.File",$fileName);
        return (int) java_values($file->length());
    }
}